addons/google_account/google_account.py

```diff
@@ -89,14 +89,17 @@ def _get_google_token_json(self, cr, uid, authorize_code, service, context=None)
 
         headers = {"content-type": "application/x-www-form-urlencoded"}
 
-        try:
-            uri = self.get_uri_oauth(a='token')
-            data = werkzeug.url_encode(params)
+        if client_id and client_secret:
```

Comment:
For making a smaller diff, just do:
```
if not client_id or not client_secret:
    return res
```
and you can put it even higher in the code.